.github/workflows/docker_build_test_push.yml

```diff
@@ -0,0 +1,312 @@
+name: ci
+
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the master branch
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+env:
+  # - change per env
+  TF_VAR_project_name: fffeastcd
+
+  # - account used to access the internal pypi repository
+  TF_VAR_pypi_user: ci
+
+  # - optionally change per env
+  TERRAFORM_BACKEND_STORAGE_ACCOUNT_RESOURCE_GROUP_NAME: fffeasttf
+  # - change per env
+  # - manually crate this storage account in the
+  #   resource group TERRAFORM_BACKEND_STORAGE_ACCOUNT_RESOURCE_GROUP_NAME
+  TERRAFORM_BACKEND_STORAGE_ACCOUNT_NAME: fffeastcdtf
+  # - manually create this container in the created storage account
+  TERRAFORM_BACKEND_CONTAINER_NAME: terraform
+
+  # no need to change those
+
+  # Base name for images published into ACR (with -core, -serving suffix)
+  CONTAINERREGISTRY_IMAGENAMEBASE: fffeast
+
+  # Blob names used to store state in Terraform backend
+  TERRAFORM_BACKEND_BLOB_PATH_INFRA: infra
+  TERRAFORM_BACKEND_BLOB_PATH_APP: app
+
+  # Name of end-to-end test pod
+  E2E_TEST_POD: e2e-test
+
+  DOCKER_COMPOSE: docker-compose -f infra/docker-compose/docker-compose.yml -f infra/docker-compose/docker-compose.online.yml -f infra/docker-compose/docker-compose.databricks.yml
+
+jobs:
+
+  build-and-test:
+
+    # Run the job on a self-hosted agent VM. The VM will cache Docker layers, which considerably speeds up the build.
+    runs-on: self-hosted
+    steps:
+
+    # Delete any files left from the previous build, including files owned by root created by Docker in mounted volumes.
+    - name: Clean current directory
+      run: shopt -s dotglob; sudo rm -rf *
+
+    # Checkout sources from Git
+    - uses: actions/checkout@v2
+
+    - name: Setup Terraform
+      uses: hashicorp/setup-terraform@v1
+      with:
+        terraform_version: 0.12.24
+        terraform_wrapper: false
+
+    - name: Set Terraform variables infra
+      run: |
+        echo ::set-env name=ARM_CLIENT_ID::${{ secrets.ARM_CLIENT_ID }}
+        echo ::set-env name=ARM_CLIENT_SECRET::${{ secrets.ARM_CLIENT_SECRET }}
+        echo ::set-env name=ARM_SUBSCRIPTION_ID::${{ secrets.ARM_SUBSCRIPTION_ID }}
+        echo ::set-env name=ARM_TENANT_ID::${{ secrets.ARM_TENANT_ID }}
+
+    - name: Terraform init infra
```

Comment:
Why introduce terraform given the other myriad of tools already used?

Reply:
Sorry for any confusion about what's included in this PR. Terraform, Docker GitHub Actions, etc code as been left in to demonstrate our internal build + test process and help understand what needs to be integrated into the main Feast testing strategy. We don't really intend on including that in final PRs.